src/test/lambda/models/samTemplates.test.ts

```diff
@@ -0,0 +1,67 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as assert from 'assert'
+import {
+    getSamCliTemplateParameter,
+    getSamTemplateWizardOption,
+    getTemplateDescription,
+    helloWorldOption,
+    repromptUserForTemplate,
+    validTemplateOptions
+} from '../../../lambda/models/samTemplates'
+import { assertThrowsError } from '../../../test/shared/utilities/assertUtils'
+
+import { samLambdaRuntimes } from '../../../lambda/models/samLambdaRuntime'
+
+describe('getSamTemplateWizardOption', () => {
+    it('should successfully return available templates for specific runtime', () => {
+        for (const runtime of samLambdaRuntimes.values()) {
+            const result = getSamTemplateWizardOption(runtime)
+            switch (runtime) {
+                case 'python3.6':
+                case 'python3.7':
+                case 'python3.8':
+                    assert.deepStrictEqual(
+                        result,
+                        validTemplateOptions,
+                        'Event bridge app supports all valid template options'
+                    )
+                    break
+                default:
+                    assert.deepStrictEqual(
+                        result,
+                        helloWorldOption,
+                        'Rest of the runtimes support hello-world template only'
+                    )
+                    break
+            }
+        }
+    })
+})
+
+describe('getSamCliTemplateParameter', () => {
+    it('should successfully return template values used by sam cli', () => {
+        for (const template of validTemplateOptions.values()) {
+            const result = getSamCliTemplateParameter(template)
+            assert.ok(result, `Template name on the wizard : ${template}, sam cli parameter value : ${result}`)
+        }
+    })
+
+    it('should return error if the template option is not valid', async () => {
+        const errorMessage = `Template ${repromptUserForTemplate} is not supported for sam application flow`
```

Comment:
Not 100% sure why you're using `repromptUserForTemplate` as your string here. I'd just create a dummy const for this test to cut down on imports.

Reply:
So, `getSamCliTemplateParameter`  method takes SamTemplate object as a parameter and in my SamTemplate type definition I do not have string,  only 3 valid template strings and one invalid.

Reply:
It's a bit of a strange error in that case, I'm assuming a user is never meant to receive it? It's a bit of a nit, but maybe refocus the error message.